nnstreamer/tensor_filter/tensor_filter_nntrainer.cc

```diff
@@ -210,8 +218,22 @@ static int nntrainer_setInputDim(const GstTensorFilterProperties *prop,
     static_cast<NNTrainerInference *>(*private_data);
   g_return_val_if_fail(prop && nntrainer && in_info && out_info, -EINVAL);
 
-  auto num_input = in_info->num_tensors;
-  g_return_val_if_fail(num_input != 0, -EINVAL);
+  auto model_inputs = nntrainer->getInputDimension();
+
+  /// if num_input is zero, we set a default input shape with batch of 1.
+  if (in_info->num_tensors == 0) {
```

Comment:
`in_info->num_tensors` indicates the number of inputs to be given to the model, and not the batch size.
Refer https://github.com/nnstreamer/nnstreamer/blob/c8d76bb71bd53dc16d4d48512a654da656aa3db0/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc#L663.